packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -202,56 +258,64 @@ export class WaferMap extends FoundationElement {
     }
 
     private quadrantChanged(): void {
-        this.queueRender();
-    }
-
-    private orientationChanged(): void {
-        this.queueRender();
+        this.waferMapUpdateTracker?.track('quadrant');
```

Comment:
Can you remove all the conditional checks now? i.e. `this.waferMapUpdateTracker.track('quadrant');`

Reply:
I left these checks here because there are cases when the waferMapUpdateTracker is still undefined and a property changes. for example, the first attribute gets its default value set which triggers a change event before the waferMapUpdateTracker is initialized. Maybe there is something I do not understand about the initialization and connected callback of the component, but without these the unit tests are failing. Can you also check this behavior? Maybe we can have a call to find  a solution to this when you have the time

Reply:
I think I see what is happening and yea it's a bit tricky:

So this code:
```ts
class Ex{
   @observable
   public myProp = 'a';

  private updateTracker = new UpdateTracker(this);
  
  myPropChanged() {
    updaterTracker.doSomething();
  }
}
```

When compiled desugars to something like:
```ts
class Ex{
   get myProp {
       return this._myProp;
   }
   set myProp(v) {
      this._myProp = v;
      this.myPropChanged();
  }  
  myPropChanged() {
    updaterTracker.doSomething();
  }
  constructor() {
      this.myProp = 'a';
      this.updateTracker = new UpdateTracker(this);
  }
}
```

So the issue here is literally the order of instantiation. The property assignment runs first causing the changed callback to run before the update tracker was constructed.

Instead, you need the updateTracker to be created before any of those observable property assignments run.

```ts
class Ex{
   /* @internal */
   public updateTracker = new UpdateTracker(this);

   @observable
   public myProp = 'a';

  myPropChanged() {
    updaterTracker.doSomething();
  }
}
```

Making sure the objects needed in the changed handlers are created first is also the approach used in TableColumns (which because of our lint rules encourages making it public and marking it internal) See:
https://github.com/ni/nimble/blob/1a5679389fe3439214ff0eb8d0029e7e6e755a61/packages/nimble-components/src/table-column/base/index.ts#L25

Reply:
Hopefully that clears it up and we can initialize the update tracker first but feel free to reach out if you need to discuss / have Qs